forge-gui/res/cardsfolder/upcoming/gilraen_dunedain_protector.txt

```diff
@@ -0,0 +1,14 @@
+Name:Gilraen, Dunedain Protector
+ManaCost:2 W
+Types:Legendary Creature Human Noble
+PT:2/3
+A:AB$ ChangeZone | Cost$ 2 T | ValidTgts$ Creature.Other+YouCtrl | Origin$ Battlefield | Destination$ Exile | TgtPrompt$ Select another target creature you control | SubAbility$ DBReturn | SpellDescription$ Exile another target creature you control. You may return that card to the battlefield under its owner's control. If you don't, at the beginning of the next end step, return that card to the battlefield under its owner's control with a vigilance counter and a lifelink counter on it.
+SVar:DBReturn:DB$ ChangeZone | Defined$ Targeted | Origin$ All | Optional$ True | Destination$ Battlefield | SubAbility$ DBDelayedTrigger
+SVar:DBDelayedTrigger:DB$ DelayedTrigger | RememberObjects$ Targeted | Mode$ Phase | Phase$ End Of Turn | Execute$ TrigReturn | SpellDescription$ If you don't, at the beginning of the next end step, return that card to the battlefield under its owner's control with a vigilance counter and a lifelink counter on it.
```

Comment:
this looks like the delayed trigger would always fire, that's wrong

Reply:
can't figure this one out, only thing that might work is a storevar with rememberchanged on line 6 ?

Reply:
You are remembering if it gets returned, then your condition checks if something got remembered:
that's the exact opposite of what the card wants

Reply:
I'd say that most likely didn't work before either :/
Check _Semester's End_ for pointers

Reply:
finally 
![image](https://github.com/Card-Forge/forge/assets/67333662/b813234e-434f-4c47-8586-fe614075f0b7)
🥳

Reply:
Yes but your copy&paste still needs work:
- card isn't called Semester's End
- nothing says it should only get counters if it's still a creature

Reply:
thanks again, only thing that might still need work is that the cleanup doesn't happen at the right time ?
![image](https://github.com/Card-Forge/forge/assets/67333662/981cd202-bd1d-46c3-b197-1859abc4e6e8)

Reply:
fixed now